### PR TITLE
[FIX] Payment Error State 

### PR DESCRIPTION
**Pull request by iamspathan**
FIX Payment Error State 

Changes: Added Payment Failed Fragment

Screenshot: 
gif

### CODE REVIEW

File: app/src/main/java/com/codingblocks/cbonlineapp/course/checkout/CheckoutActivity.kt [hunk 1]
```diff
@@ -100,4 +104,15 @@ class CheckoutActivity : BaseCBActivity(), PaymentResultWithDataListener {
             }
         }
     }
+
+    override fun onSaveInstanceState(outState: Bundle, outPersistentState: PersistableBundle) {
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/course/checkout/CheckoutActivity.kt]: remove not needed

**Review by aggarwalpulkit596 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/codingblocks/cbonlineapp/util/extensions/ViewExtensions.kt [hunk 2]
```diff
@@ -121,12 +121,14 @@ fun AppCompatActivity.replaceFragmentSafely(
...
     @AnimRes popEnterAnimation: Int = 0,
-    @AnimRes popExitAnimation: Int = 0
+    @AnimRes popExitAnimation: Int = 0,
+    addToStack: Boolean = false
 ) {
     val ft = supportFragmentManager
         .beginTransaction()
         .setCustomAnimations(enterAnimation, exitAnimation, popEnterAnimation, popExitAnimation)
         .replace(containerViewId, fragment, tag)
+    if (addToStack) { ft.addToBackStack("")}
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/util/extensions/ViewExtensions.kt]: Add to Stack using a tag

**Review by aggarwalpulkit596 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/codingblocks/cbonlineapp/util/extensions/ViewExtensions.kt (hunk 2 above)
Comment (iamspathan) [reply app/src/main/java/com/codingblocks/cbonlineapp/util/extensions/ViewExtensions.kt]: changes made in last commit

File: app/src/main/java/com/codingblocks/cbonlineapp/course/checkout/CheckoutActivity.kt (hunk 1 above)
Comment (iamspathan) [reply app/src/main/java/com/codingblocks/cbonlineapp/course/checkout/CheckoutActivity.kt]: changes made in last commit

**Review by aggarwalpulkit596 — APPROVED** [review]

**Pull request merged**